aws/tests/test_publish.py

```diff
@@ -20,13 +20,19 @@ def mdf_other_user_datset(mdf):
 def dyanmo_record_version(automate_record):
     assert automate_record['is_test']
 
+
 @then(parsers.parse("the only data destinations should be {destination}"))
-def dyanmo_record_version(automate_record):
+def dyanmo_record_version(destination, automate_record):
     dest_org = automate_record['organization']
     print("dest org", dest_org)
     assert len(dest_org.data_destinations) == 1
 
 
+@then(parsers.parse("the dataset's domain should be '{domain}'"))
+def check_dataset_domain(domain, automate_record):
+    assert automate_record['mdf_rec']['mdf']['domains'][0] == domain
```

Comment:
@blaiszik this is the heart of the change... test that the search record has a property called `domains` and it gets set to the correct value